src/main/java/com/jcabi/s3/AwsOcket.java

```diff
@@ -198,6 +200,8 @@ public void write(
             );
         } catch (final AmazonServiceException ex) {
             throw new IOException(ex);
+        } catch (final InterruptedException ex) {
```

Comment:
@caarlos0 It will not compile in Java 5,6. It is bad idea. What you sink about java 5 and 6 for this project?